pkg/tfsandbox/plan.go

```diff
@@ -66,12 +67,58 @@ func (t *Tofu) planWithOptions(ctx context.Context, logger Logger, refreshOnly b
 		return nil, fmt.Errorf("error running plan: %w", err)
 	}
 
-	// NOTE: the recommended default from terraform-json is to set JSONNumber=true
-	// otherwise some number values will lose precision when converted to float64
-	plan, err := t.tf.ShowPlanFile(ctx, planFile, tfexec.JSONNumber(true))
-	if err != nil {
-		return nil, fmt.Errorf("error running show plan: %w", err)
+	var wg sync.WaitGroup
```

Comment:
Go makes this kind of parallelism really awkward. You have a lot of machinery in use here, it's probably fine but heavy. A bit of Go golf, you really need only one WaitGroup or one chan, and can do a fork-join pattern. Can also leave micro-optimization for later with a comment, sorry for picking here.

```go
func fx() (int, error) {
	return 2, fmt.Errorf("f-failed")
}

func gx() (int, error) {
	return 1, fmt.Errorf("g-failed")
}

func par() (int, error) {

	var (
		fr   int
		ferr error
		fch  chan bool = make(chan bool)
	)

	// fork
	go func() {
		defer close(fch)
		fr, ferr = fx()
	}()

	gr, gerr := gx()

	// join
	<-fch

	err := errors.Join(ferr, gerr)
	if err != nil {
		return 0, err
	}

	return fr + gr, nil
}

```